### ISSUE TITLE
MacOS Mojave - ERR_PROXY_CONNECTION_FAILED 

### ISSUE THREAD

**Issue by thomasKuntermannDiva-e** [lightbody/browsermob-proxy#802]
I was facing the issue, that after updating my MacOS from 10.13.x to 10.14 (Mojave), that browsermobproxy was not able to to work with chromedriver + firefoxdriver anymore. I got the error displayed:
![44606994-f28fca00-a7bc-11e8-80dd-fc994a719b1b](https://user-images.githubusercontent.com/27360136/51318828-ad19b500-1a5b-11e9-980e-dfe46bbe6c55.png)

that was happening, when working with an active VPN connection to my companies network.

Solution for that: 
- edit /etc/hosts via "sudo nano /etc/hosts"
- add your Mac HOSTNAME (with ".local") to the entries "127.0.0.1" and "::1"
- press CTRL + o and leave with CTRL + x
- restart DNS serice: "sudo killall -HUP mDNSResponder"

**
127.0.0.1       localhost,mbpro-username.local
255.255.255.255 broadcasthost
::1             localhost,mbpro-username.local
**

**Comment by Mandeep-S** [comment]
@tkuntermannpv why is it open issue ? or you had resolved it with above steps ?

**Comment by thomasKuntermannDiva-e** [comment]
@Mandeep-S i opened this issue and left it open, so that you may find a solution for that without needing my workaround to be applied. Before the upgrade to MacOS Mojave, it worked without my described workaround.
Don't know what caused the described issue, but maybe you find out, if you can fix this within browsermob-proxy?

**Comment by Mandeep-S** [comment]
@tkuntermannpv , I followed your instructions and it worked for me. Thanks for the solution. :)

**Comment by thomasKuntermannDiva-e** [comment]
:) nice! i just have no idea, why it works currently only that way :(

**Comment by Mandeep-S** [comment]
@tkuntermannpv I have another system Mac High Sierra (version 10.13.3) and I am getting ERR_CONNECTION_TIMED_OUT there with same code ? Any idea or fix for that ? I tried above solution but it didnt worked there.

**Comment by thomasKuntermannDiva-e** [comment]
are you using a VPN client while executing browsermob-proxy @Mandeep-S ? if yes, try to turn it off (before upgrading to MacOS Mojave i had this issue only, when using a VPN client to get into my companies network, what was not necessary for accessing the SystemUnderTest).
My VPN Client is "Cisco AnyConnect" (don't know if this makes any difference for this issue, but mention it for documentation ;) )

**Comment by Mandeep-S** [comment]
no, I am not using a VPN client. The system is actually company's system where I am trying to run.

**Comment by thomasKuntermannDiva-e** [comment]
hm, you said in https://github.com/lightbody/browsermob-proxy/issues/802#issuecomment-456815099, that the solution worked for you? have you tried this on 2 different MacOS versions until now?

**Comment by Mandeep-S** [comment]
yes, In Mojave its working but not in High Seirra

**Comment by thomasKuntermannDiva-e** [comment]
good hint, then the issue must be introduced with MacOS 10.13.3, because i tried on a Virtual Machine last week with MacOS 10.13.2 and there i had no issue at all (without adapting the /etc/hosts fix).
Maybe 10.13.3 is the error-causing OS version. 
I suggest to not run browsermob-proxy on that version, as long as nobody else has any idea on that

**Comment by s2oBCN** [comment]
tkuntermannpv. Thank you for sharing your workaround !! This worked for me :)

**Comment by lindaxuxlj** [comment]
Thanks tkuntermannpv. Thank you for sharing your workaround on Mac !!  Do u know how to solve this problem on windows?

**Comment by thomasKuntermannDiva-e** [comment]
@lindasunflower unfortunately not. But i expect, that Windows may have a quite similar setup possibility. it was even hard to find a solution for MacOS :D 
If you find a solution, it would be great to share it here too.

Good luck ;)
